improver/nbhood/recursive_filter.py

```diff
@@ -419,7 +438,17 @@ def process(
 
             recursed_cube.append(new_cube)
 
+            if mask_cube is not None:
+                new_cube.data = np.ma.MaskedArray(new_cube.data, mask=mask_cube.data)
+
```

Comment:
Not sure why these lines have been duplicated.
```suggestion
```

Reply:
Removed duplicate line.